packages/dds/sequence/src/test/attribution/sharedString.attribution.spec.ts

```diff
@@ -0,0 +1,697 @@
+				factory: (operations: Operation[]) => createSharedString(
+					makeRandom(0),
+					generatorFromArray(operations),
+					(runtime) => chainEncoders(
+						new AttributorSerializer(
+							(entries) => new OpStreamAttributor(runtime.deltaManager, runtime.getAudience(), entries),
+							noopEncoder
+						),
+						makeGzipEncoder(),
+					)
+				),
+				filename: "attributor-gzip-compression-snap.json",
+			},
+			{
+				name: "OpStreamAttributor",
+				factory: (operations: Operation[]) => createSharedString(
+					makeRandom(0),
+					generatorFromArray(operations),
+					(runtime) => chainEncoders(
+						new AttributorSerializer(
+							(entries) => new OpStreamAttributor(runtime.deltaManager, runtime.getAudience(), entries),
+							deltaEncoder
+						),
+						makeGzipEncoder(),
+					)				),
+				filename: "attributor-gzip-and-delta-snap.json",
+			},
+		];
+
+		it.skip("Generate a new document", async () => {
+			const paths = getDocumentPaths("basic");
+			const attributionlessGenerator = chain(
+				take(100, makeOperationGenerator({ validateInterval })),
+				generatorFromArray<Operation, FuzzTestState>([{ type: "synchronize" }]),
+			);
+
+			const { generator, operations } = spyOnOperations(attributionlessGenerator);
+			createSharedString(makeRandom(0), generator);
+			writeJson(paths.operations, operations);
+
+			await Promise.all(
+				dataGenerators.map(async ({ filename, factory }) => {
+					const summary = await summaryFromState(factory(operations));
+					writeJson(path.join(paths.directory, filename), summary);
+				}),
+			);
+		});
+
+		const documents = getDocuments();
+		for (const document of documents) {
+			describe(`document name: ${document} has an up-to-date`, () => {
+				let paths: TestPaths;
+				let operations: Operation[];
+				before(() => {
+					paths = getDocumentPaths(document);
+					operations = readJson(paths.operations);
+				});
+
+				for (const { filename, factory } of dataGenerators) {
+					it(`snapshot at ${filename}`, async () => {
+						const expected = readJson(path.join(paths.directory, filename));
+						const actual = await summaryFromState(factory(operations));
+						assert.deepEqual(actual, expected);
+					});
+				}
+			});
+		}
+
+		// Note: to see output, FLUID_TEST_VERBOSE needs to be enabled. Using the `test:mocha:verbose` script is
+		// sufficient to do so.
+		it("generate snapshot size impact report", async () => {
+			const table = new DataTable<ISummaryTree>(dataGenerators.map(({ name }) => name));
+			for (const docName of documents) {
+				const paths = getDocumentPaths(docName);
+				const operations: Operation[] = readJson(paths.operations);
+				const data = await Promise.all(
+					dataGenerators.map(async ({ factory }) => summaryFromState(factory(operations))),
+				);
+				table.addRow(docName, data);
+			}
+
+			table.log(getSummaryLength);
+		});
+	});
+
+	describe.skip("using sample real documents", () => {
```

Comment:
I found this useful for prototyping, not sure whether we want to check it in as part of this PR, check it in later with some cleanup, or not check it in at all. It's a variant of the above suite which--rather than generate random documents--is able to take an actual Scriptor doc, strip its attribution info, and make analogous comparisons. If we do want to do this kind of thing, it might make more sense to put it in e2e tests. Open to thoughts in the realm.

Reply:
kind of leaning toward pushing a branch with this code and filing some work to think about what e2e test coverage we want for real documents + attribution.

Reply:
filed https://dev.azure.com/fluidframework/internal/_workitems/edit/2725 to track this with some pointers to this code. going to delete from this PR.